source/Preprocessor.cpp

```diff
@@ -523,18 +525,19 @@ MacroActualArgumentListSyntax* Preprocessor::handleTopLevelMacro(Token directive
     expandedTokens.clear();
     for (uint32_t i = 0; i < tokens.count(); i++) {
         Token newToken;
-        Token token = tokens[i];
 
         // replace intrinsic macros before we do anything else
-        // TODO: fill these in
+        Token token = tokens[i];
+
         if (token.kind == TokenKind::IntrinsicFileMacro) {
             auto info = alloc.emplace<Token::Info>(token.trivia(), "", token.location(), 0);
-            info->extra = "";
+            info->extra = sourceManager.getFileName(token.location());
             token = Token(TokenKind::StringLiteral, info);
-        }
-        else if (token.kind == TokenKind::IntrinsicLineMacro) {
+        } else if (token.kind == TokenKind::IntrinsicLineMacro) {
             auto info = alloc.emplace<Token::Info>(token.trivia(), "", token.location(), 0);
-            info->setNumInfo(SVInt(0));
+            uint64_t lineNum = sourceManager.getLineNumber(token.location());
+            // sourceManager gives a 0-indexed line number
```

Comment:
I am pretty sure it is correct, i.e I very consistently got correct values for __LINE__ that were one less than the correct value, including a value of 0 if it was on the first line. I didn't dig into DiagnosticWriter to see if that made sense, but it is likely that everything would be better if the "+1" were in getLineNumber